### PR TITLE
DOC: rename Usage Guide to Basic Usage and update

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

Renames Usage Guide tutorial  to Basic Usage (tutorials/introductory/usage.py).  This adds _extensively_ to this guide to provide a short overview of how to use Matplotlib, with links to other tutorials or examples where appropriate or necessary.  This is meant to be on par with the pyplot tutorial, but doesn't follow it slavishly.

### History

This page used to have a bunch of explanatory material.  A lot of this was moved in #21335 to `/docs/users` leaving this page quite thin.  There is already the pyplot tutorial, and the "lifecycle of a plot" tutorial, but the pyplot tutorial is not the style we'd prefer people to use, and "lifecycle" goes deep in a couple of directions.  (Actually, I'd like to remove that tutorial as well, or make it more "scientific" by not just making a bar plot of all things, but that is lower priority)

### Goal

Not completeness, but enough ideas of what the library can do to point users to more complete resources.  

Use default styles wherever possible.  Only exception are the figure sizes, which are usually unsuitable for a website tutorial (in my opinion).

If there is a major _part_ of the library for _basic_ usage, feel free to comment.  

### Outline:

Basic usage
- plot
- anatomy
- Note on coding styles
- styling artists
  - colors
  - linestyles
  - markerstyles	
- labelling plots
  - xlabel, ylabel and title
  - axis: ticks and ticklabels (locators and formatters)
    - units: categorical and dates
  - text
  - annotations 
  - legends
- colormapping data
   - basic and colormaps
   - vmin, vmax
   - norms
   - colorbars
- arranging axes
   - subplots
   - subplot_mosaic

[NOTE: depends on #21631 - I'll rebase when that is done.  Only file to change is `tutorials/introductory/usage.py`]

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by jklymak** [comment]
rebased on main...

File: tutorials/introductory/usage.py [hunk 1]
```diff
@@ -59,48 +42,44 @@
...
 #    fig = plt.figure()  # an empty figure with no Axes
 #    fig, ax = plt.subplots()  # a figure with a single Axes
 #    fig, axs = plt.subplots(2, 2)  # a figure with a 2x2 grid of Axes
 #
-# It's convenient to create the axes together with the figure, but you can
-# also add axes later on, allowing for more complex axes layouts.
+# It is often convenient to create the axes together with the figure, but you
+# can also manually add axes later on.  Note that many
+# :doc:`Matplotlib backends </users/explain/backends>` support zooming and
+# panning on figure windows.
```
Comment (tacaswell) [inline tutorials/introductory/usage.py:57]:
```suggestion
# panning on Axes.
```

File: tutorials/introductory/usage.py [hunk 2]
```diff
@@ -120,25 +99,40 @@
...
+##############################################################################
 # .. _coding_styles:
 #
-# The object-oriented interface and the pyplot interface
-# ======================================================
+# Coding styles
+# =============
+
+# The object-oriented and the pyplot interfaces
+# ---------------------------------------------
```
Comment (tacaswell) [inline tutorials/introductory/usage.py:135]:
```suggestion
# The explicit (object-oriented) and the implicit (pyplot) interfaces
# -------------------------------------------------------------------
```

File: tutorials/introductory/usage.py [hunk 3]
```diff
@@ -175,77 +170,373 @@
...
-# again with different data sets, use the recommended signature function below.
+# If you need to make the same plots over and over again with different data sets,
+# or want to easily wrap Matplotlib methods, use the recommended signature function
+# below.
 
 
 def my_plotter(ax, data1, data2, param_dict):
     """
-    A helper function to make a graph
+    A helper function to make a graph.
```
Comment (tacaswell) [inline tutorials/introductory/usage.py:199]: I have mixed feeling about dropping the docstring.  On one hand, I see the reason to make it terser, on the other hand we should model best practices.

Comment (jklymak) [reply tutorials/introductory/usage.py:199]: I don't feel strongly about this, and am happy to add it back in.  However, I think for a little in-script utility its OK to not have a numpydoc compliant docstring, and I found it distracting for a tutorial.

File: tutorials/introductory/usage.py [hunk 4]
```diff
@@ -175,77 +170,373 @@
...
+mu, sigma = 100, 15
+x = mu + sigma * np.random.randn(10000)
+fig, ax = plt.subplots(figsize=(5, 2.7), constrained_layout=True)
+# the histogram of the data
+n, bins, patches = ax.hist(x, 50, density=1, facecolor='C0', alpha=0.75)
+
+ax.set_xlabel('Smarts')
+ax.set_ylabel('Probability')
+ax.set_title('Histogram of IQ')
+ax.text(60, .025, r'$\mu=100,\ \sigma=15$')
```
Comment (tacaswell) [inline tutorials/introductory/usage.py:297]:
```suggestion
ax.set_xlabel('Value')
ax.set_ylabel('Probability')
ax.set_title(r'Histogram of normal distribution')
ax.text(60, .025, r'$\mu=100,\ \sigma=15$')
```

**Review by tacaswell — APPROVED** [review]
I think this a big improvement.

I left one additional comment changing the text in an example (IQ can be controversial and is non-material to the point we are trying to make so might as well avoid courting controversy), but  won't block merging over that.

@jklymak can self-merge either taking the suggestion or not.

**Comment by jklymak** [comment]
Yeah that one was left over from before so I did t change it.   It I wondered about it as well and am happy to  switch to aardvark length or something.  

We actually have a gender one somewhere which strikes me as even more potentially contentious - i will try and remember to open an issue for that

File: tutorials/introductory/usage.py (hunk 4 above)
Comment (jklymak) [reply tutorials/introductory/usage.py]: Changed to aardvark length (with a caveat that the distribution was not real, though I changed the mean to be correct for aardvarks).

**Pull request merged**